### ISSUE TITLE
Appease JSCS

### ISSUE THREAD

**Issue by zachwills** [reaktivstudios/locomotive#41]
JSCS doesn't like most of the JS files. Can run JSCS auto fix for some of them, will need to manually update others.

**Comment by JasonHoffmann** [comment]
Added an ESLint config to the branch 'amend-jscs'. Will need to fix style issues before submitting a PR but wanted to make sure that this is the configuration everyone wanted. Uses the rules from the existing codebase, plus a recommended set of rules from ESLint and React.

**Comment by jjeaton** [comment]
@JasonHoffmann it's OK to open a WIP PR, even before the style issues are fixed.

I am interested in developing our internal eslint standards, as there isn't a standard WP eslintrc at the moment. Some options:

https://github.com/ntwb/eslint-config-wordpress
https://urbangiraffe.com/2015/02/23/wordpress-javascript-code-standard-eslint/

TL;DR: React standards are cool, but as most of our projects are going to be WP, let's see how we can merge these standards.

**Comment by JasonHoffmann** [comment]
I've combined rulesets from a few different places to identify the best practices and mimic WP:

```
  "rules": {
        "array-bracket-spacing": [ "error", "always" ],
        "brace-style" : [ "error", "1tbs" ],
        "camelcase": "error",
        "computed-property-spacing": [ "error", "always" ],
        "curly" : "error",
        "eol-last": "error",
        "eqeqeq": "error",
        "indent": [ 1, "tab" ],
        "keyword-spacing": "error",
        "linebreak-style": [ "error", "unix" ],
        "no-caller": "error",
        "no-eq-null": "error",
        "no-else-return": "warn",
        "no-mixed-spaces-and-tabs": [1, "smart-tabs"],
        "no-multi-spaces": "error",
        "no-nested-ternary": "error",
        "no-shadow": "error",
        "no-spaced-func": "error",
        "no-trailing-spaces": "error",
        "no-unused-expressions": "error",
        "no-unused-vars": "error",
        "object-curly-spacing": [ "error", "always" ],
        "quotes": [ "error", "single" ],
        "react/no-direct-mutation-state" : "off",
        "react/jsx-curly-spacing": [ 1, "always" ],
        "react/jsx-space-before-closing" : "warn",
        "semi": [ "error", "always" ],
        "semi-spacing": "warn",
        "space-before-blocks" : "error",
        "space-before-function-paren" : "error",
        "space-in-parens" : [ "error", "always" ],
        "wrap-iife": [ "error", "any" ]
    }
```

#### Highlights:
- Tabs instead of spaces
- Snakecase instead of camelcase
- Space around and inside parentheses, functions, curly braces
- Semicolons required
- No whitespace

There are some changes that need to be made to get this code compliant, but wanted to get a consensus on these rules

**Comment by jjeaton** [comment]
I like where this is going. The one I may want to change is snake case. I
like camelCase in JS. It seems to fit better. What do you think? @mkellyxp,
your thoughts as well?

**Comment by JasonHoffmann** [comment]
One issue I ran into is that object properties sometimes need to be snake_case in order to match submissions to the backend. Fortunately, ESLint allows for that, so I've allowed it on object properties but not in variables or functions.

I've pushed up a version of the code that has all the issues fixed with the new config, so you can take a look.
